### PR TITLE
pixman: patch in build() instead of source() + bump msys2 + del fPIC if shared + fix hooks

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **pixman/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Also for consistency across recipes:
- rename source subfolder after download/unarchive
- delete cppstd and libcxx in `configure()` instead of `config_options()`

### CODE REVIEW

**Comment by SpaceIm** [comment]
@ericLemanissier I saw that test_package doesn't call pixman symbols if windows, do you know why? Would there be a way to have a minimal working test_package on windows?

**Comment by ericLemanissier** [comment]
I don't know, but I just tested without the #ifdefs in debug and release, it passed. It's worth trying !

File: recipes/pixman/all/conanfile.py [hunk 1]
```diff
@@ -17,40 +16,49 @@ class PixmanConan(ConanFile):
...
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
-        if self.settings.os == 'Windows':
-            del self.options.fPIC
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("pixman can only built as static library for Windows")
```
Comment (madebr) [inline recipes/pixman/all/conanfile.py:36]: Grammar nazi, to your service :smile: 
```suggestion
            raise ConanInvalidConfiguration("pixman can only be built as a static library on Windows")"
```

File: recipes/pixman/all/conanfile.py [hunk 2]
```diff
@@ -91,11 +96,11 @@ def package(self):
         else:
             autotools = self._configure_autotools()
             autotools.install()
-        la = os.path.join(self.package_folder, "lib", "libpixman-1.la")
-        if os.path.isfile(la):
-            os.unlink(la)
-        self.copy(os.path.join(self._folder, 'COPYING'), dst='licenses')
-        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
+            tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
```
Comment (madebr) [inline recipes/pixman/all/conanfile.py:99]: While you're at it, I think you forgot to convert these to double quotes.

File: recipes/pixman/all/conanfile.py [hunk 3]
```diff
@@ -33,7 +33,7 @@ def configure(self):
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
         if self.settings.os == "Windows" and self.options.shared:
-            raise ConanInvalidConfiguration("pixman can only built as static library for Windows")
+            raise ConanInvalidConfiguration("pixman can only be built as a static library on Windows")"
```
Comment (SpaceIm) [inline recipes/pixman/all/conanfile.py:36]:
```suggestion
            raise ConanInvalidConfiguration("pixman can only be built as a static library on Windows")
```

**Review by prince-chrismc — DISMISSED** [review]

**Comment by ericLemanissier** [comment]
@SpaceIm you can merge this one in your PR if you want https://github.com/conan-io/conan-center-index/pull/3258

**Comment by SpaceIm** [comment]
@ericLemanissier done

File: recipes/pixman/all/conanfile.py (hunk 2 above)
Comment (SSE4) [reply recipes/pixman/all/conanfile.py:99]: I doubt it's needed, only adds additional noise

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**